momento-sdk/src/main/java/momento/sdk/ScsTopicGrpcStubsManager.java

```diff
@@ -100,8 +135,30 @@ PubsubGrpc.PubsubStub getNextUnaryStub() {
   }
 
   /** Round-robin subscribe stub. */
-  PubsubGrpc.PubsubStub getNextStreamStub() {
-    return streamStubs.get(streamIndex.getAndIncrement() % this.numStreamGrpcChannels);
+  StreamStubWithCount getNextStreamStub() {
+    // First check if there's available capacity on any of the stubs
+    int totalActiveSubscriptions = 0;
+    for (StreamStubWithCount stubWithCount : streamStubs) {
+      totalActiveSubscriptions += stubWithCount.getCount();
+    }
```

Comment:
Do we need to tally this up given we check each channel below individually for capacity?